server/src/server.rs

```diff
@@ -1049,6 +1087,15 @@ where
 
 		request.extensions_mut().insert::<ConnectionId>(conn.conn_id.into());
 
+		if let Some(ip_addr) = self.inner.ip_addr {
+			// Only insert the remote address if it's not already set.
+			// We expect servers deployed behind a reverse proxy to set the remote address
+			// themselves otherwise the remote address will be the address of the reverse proxy.
+			if request.extensions().get::<IpAddr>().is_none() {
+				request.extensions_mut().insert(ip_addr);
```

Comment:
Added this for better logging but I can revert this

Reply:
I'm probably being dumb but: how does setting thr IP address of the machine running this server help with logging? To help differentiate when logs from lots of servers are mingled together?

Reply:
It's useful for debug logs if your server has lots connections and maybe only certain peers/ip addrs are slow but again I think it's better to revert this from this PR..